### ISSUE TITLE
"Recover after Nodes / Time" logic and naming is confusing

### ISSUE THREAD

**Issue by polyfractal** [elastic/elasticsearch#7389]
The current recover after node/time logic is very easily to get wrong.  The logic today is:
- `recover_after_nodes` (and friends) are hard limits.  The cluster _will not_ recover unless these conditions are met.
- Once the `recover_after_*` conditions are met, the cluster waits for `gateway.expected_nodes` OR `gateway.recover_after_time`, whichever comes first.

This is a confusing sequence of logic, since `gateway.recover_after_time` looks like a "failsafe" that will start recovery after X period of time, regardless of the other settings.  In reality, it is only invoked if the first `recover_after_*` settings are satisfied.

At a minimum, the docs should spell this out more clearly.  Better, we should probably do some renaming.  Perhaps:
- `recover_after_nodes` becomes `minimum_node_count` ?
- `gateway.recover_after_time` becomes `gateway.wait_for_expected` ?

And there should probably be a new setting, `force_recover_after_time` which will force recovery regardless of any other setting, in case people want a nuclear failsafe option to get the cluster back online.

**Comment by clintongormley** [comment]
@bleskes Are these setting still required with stale shard detection?  What about master election?

**Comment by bleskes** [comment]
@clintongormley we don't indeed need them for shard security. They are however nice as they prevent too much shard relocation upon start up - waiting on nodes to join means the master will be more likely to find existing shard copies and assign replicas there rather than making fresh copies.

**Comment by clintongormley** [comment]
OK - given the diminishing role of these settings and the fact that nobody else has complained in the last two years, I'm going to close this as won't fix
